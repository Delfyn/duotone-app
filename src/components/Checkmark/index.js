/* @flow */

import React from 'react';

type Props = {
  className?: string,
  color?: string,
  size?: number,
  onClick?: Function,
};

const Checkmark = ({
  className = '',
  color = '#333',
  size = 40,
  onClick,
}: Props) => (
  <svg
    className={className}
    onClick={onClick}
    width={size}
    height={size}
    viewBox={`0 0 ${size} ${size}`}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <circle id="a" cx={size / 2} cy={size / 2} r={size / 2} />
      <mask id="b" x="0" y="0" width={size} height={size} fill="#fff">
        <use xlinkHref="#a" />
      </mask>
    </defs>
    <g fill={color} fillRule="evenodd">
      <use stroke={color} mask="url(#b)" strokeWidth="4" xlinkHref="#a" />
      <path fill="#fff" d="M17.854 26.96l12.57-12.57-1.414-1.415-11.156 11.157-6.157-6.157-1.415 1.414z" />
    </g>
  </svg>
);

export default Checkmark;
