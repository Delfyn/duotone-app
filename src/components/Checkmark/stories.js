/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import Checkmark from './index';

storiesOf('Checkmark', module)
  .add('without props', () => <Checkmark onClick={action('onClick')} />)
  .add('with different colors', () => (
    <div>
      <Checkmark color="#FD3659" onClick={action('onClick')} />
      <Checkmark color="#243261" onClick={action('onClick')} />
    </div>
  ));
