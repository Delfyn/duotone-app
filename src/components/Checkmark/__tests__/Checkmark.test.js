import React from 'react';
import renderer from 'react-test-renderer';
import Checkmark from '../index';

test('Checkmark renders correctly', () => {
  const component = renderer.create(<Checkmark />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Checkmark renders colors correctly', () => {
  const component = renderer.create(<Checkmark color="#fd3659" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
