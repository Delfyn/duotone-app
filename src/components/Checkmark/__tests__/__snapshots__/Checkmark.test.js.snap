// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkmark renders colors correctly 1`] = `
<svg
  className=""
  height={40}
  onClick={undefined}
  viewBox="0 0 40 40"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <circle
      cx={20}
      cy={20}
      id="a"
      r={20}
    />
    <mask
      fill="#fff"
      height={40}
      id="b"
      width={40}
      x="0"
      y="0"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
  </defs>
  <g
    fill="#fd3659"
    fillRule="evenodd"
  >
    <use
      mask="url(#b)"
      stroke="#fd3659"
      strokeWidth="4"
      xlinkHref="#a"
    />
    <path
      d="M17.854 26.96l12.57-12.57-1.414-1.415-11.156 11.157-6.157-6.157-1.415 1.414z"
      fill="#fff"
    />
  </g>
</svg>
`;

exports[`Checkmark renders correctly 1`] = `
<svg
  className=""
  height={40}
  onClick={undefined}
  viewBox="0 0 40 40"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <circle
      cx={20}
      cy={20}
      id="a"
      r={20}
    />
    <mask
      fill="#fff"
      height={40}
      id="b"
      width={40}
      x="0"
      y="0"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
  </defs>
  <g
    fill="#333"
    fillRule="evenodd"
  >
    <use
      mask="url(#b)"
      stroke="#333"
      strokeWidth="4"
      xlinkHref="#a"
    />
    <path
      d="M17.854 26.96l12.57-12.57-1.414-1.415-11.156 11.157-6.157-6.157-1.415 1.414z"
      fill="#fff"
    />
  </g>
</svg>
`;
